cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go

```diff
@@ -37,6 +37,53 @@ const (
 	NodesKubeadmConfigClusterRoleName = "kubeadm:nodes-kubeadm-config"
 )
 
+// UploadRemoveConfiguration uploads a new configuration with the clinet removed
+func UploadRemoveConfiguration(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error {
```

Comment:
All of this needs refactoring I think. but this was the cleanest way to do this without really touching "working code" I'm very much open to ideas here.